src/main/java/ecdar/Ecdar.java

```diff
@@ -298,28 +281,26 @@ public void start(final Stage stage) {
         }));
 
         stage.setOnCloseRequest(event -> {
-            BackendHelper.stopQueries();
-
+            int status = 0;
             try {
-                backendDriver.closeAllEngineConnections();
-            } catch (IOException e) {
-                e.printStackTrace();
+                BackendHelper.clearEngineConnections();
+            } catch (BackendException e) {
+                // -1 indicates that an exception was thrown
+                status = -1;
             }
 
             Platform.exit();
-            System.exit(0);
+            System.exit(status);
```

Comment:
We should consider adding ``platform.exit`` as a [shutdown hook](https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#addShutdownHook-java.lang.Thread-) ([More doc](https://docs.oracle.com/javase/8/docs/technotes/guides/lang/hook-design.html)). Also, I think a try catch should be around ``system.exit``and that we should consider a forced exit in the catch using [``halt``](https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#halt-int-)